playbooks/robusta_playbooks/krr.py

```diff
@@ -296,6 +298,105 @@ def _generate_prometheus_secrets(prom_config: PrometheusConfig) -> List[KRRSecre
 
     return krr_secrets
 
+class ProcessScanParams(ActionParams):
+    scan_type: str
+    result: Any
+    scan_id: str
+    start_time: datetime
+    
+@action
+def process_scan(event: ExecutionBaseEvent, params: ProcessScanParams):
+    if params.scan_type.lower() != "krr":
+        logging.warning(f"Processing scans not supported for type: {params.scan_type}")
+        return
+    metadata: Dict[str, Any] = {
+        "job": {
+            "name": f"krr-job-{params.scan_id}",
+            "namespace": INSTALLATION_NAMESPACE,
+        },
+    }
+    try:
+        krr_scan = KRRResponse(**params.result)
+    except Exception as e:
+        if isinstance(e, json.JSONDecodeError):
+            logging.exception("*KRR scan job failed. Expecting json result.*")
+        elif isinstance(e, TypeError):
+            logging.error(f"*KRR scan job failed.\n Error from KRR pod:\n {params.result}.*")
+        else:
+            logging.exception(f"*KRR scan job unexpected error.*\n {e}\nReturned result from KRR: {params.result}")
+        event.emit_event(
+            "scan_updated",
+            scan_id=params.scan_id,
+            metadata=metadata,
+            state=ScanState.FAILED,
+            type=ScanType.KRR,
+            start_time=params.start_time,
+        )
+        return
+    else:
```

Comment:
No need for the else in that case since you anyway return in the except clause.

Also consider to break down to smaller function just to make it more clearer and easier to maintin. we can discuss it online if you prefer.